### PR TITLE
Skip login activity when either:

### PR DESCRIPTION
**Pull request by stevenh47**
1) current user is not null
2) no internet

### CODE REVIEW

File: app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java [hunk 1]
```diff
@@ -32,6 +34,14 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         binding = DataBindingUtil.setContentView(this, R.layout.activity_login);
         initView();
+        checkLoggedIn();
+        checkInternet();
+    }
+
+    private void checkInternet() {
+        if (!NetworkUtil.hasInternet(this)) {
+            navigateToGameList();
```
Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: If there is no logged in user, and no internet, what should we show as the username on the navigation drawer?

Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: Then there is no username we can get. Do you have any opinion what we shall put there in such case?

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: if no internet and no logged in user, we probably should let use stay on the login screen.

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: Actually, we should let user stay on login screen if there is no logged in user no matter what other condition is.

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: So I think the only condition that needs to check is if there is logged in user. If there is then go to game list. The internet check does not really needed. We will just need to handle the offline more gracefully in the game list screen.

Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: I am thinking there would be a case that Bmob's user cache being cleared (expired or some other reason), and there is no internet at that moment. In that case, blocking users from playing downloaded games is not a good experience.

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: hmm so the login is actually not an online but rather an offline check-in action. We can save the login state to a persistent place. If user click log out, we set that state to false. That way we no longer rely on the Bmob's cache feature.

Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: You mean we manually save the login information?

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: yeah. It could be just a flag.

Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: I think we should save at the username manually as well at least. Let me cancel this pull request and create a new one. 

One more thing to discuss, I notice currently we set the login activity as the main activity, so if the user logged in before, every time they start app, they will see an activity switching animation. My proposal is to set game list as the main activity, and if a user is not logged in, switch to login activity. Or is there any other solution?

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: A splash screen would be a good fit here

Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/LoginActivity.java]: SGTM, even I don't really know what that is :P

**Pull request closed without merging**